nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/script/ScriptingComponentHelper.java

```diff
@@ -265,7 +265,11 @@ public void setupVariables(final PropertyContext context) {
         scriptEngineName = context.getProperty(SCRIPT_ENGINE).getValue();
         scriptPath = context.getProperty(ScriptingComponentUtils.SCRIPT_FILE).evaluateAttributeExpressions().getValue();
         scriptBody = context.getProperty(ScriptingComponentUtils.SCRIPT_BODY).getValue();
-        modules = context.getProperty(ScriptingComponentUtils.MODULES).evaluateAttributeExpressions().asResources().flattenRecursively();
+        if (scriptEngineName.equals("python")) {
+            modules = context.getProperty(ScriptingComponentUtils.MODULES).evaluateAttributeExpressions().asResources();
```

Comment:
Why does python not "flattenRecursively()"?

Reply:
Thank for you very much, @markobean for looking at this! 

"flatten recursively"  finds all the files in a directory
This doesn't work well with python which prefers including directory names as opposed to the file names.  

More specifically, we are using sys.path.append 
https://github.com/apache/nifi/blob/rel/nifi-1.17.0/nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/script/impl/JythonScriptRunner.java#L40 
which then looks inside the directory for any needed files

It is looking for a directory name, though, so when it gets a filename (because of the "flatten recursively")
it doesn't handle it well